### ISSUE TITLE
docker_image.present always ends up failing even on correct result.

### ISSUE THREAD

**Issue by BenjaminSchubert** [saltstack/salt#42935]
### Description of Issue/Question

When running a docker_image.present state, I always end up with an error, even though it should be a success.

### Setup
(Please provide relevant configs and/or SLS files (Be sure to remove sensitive info).)

We need a salt minion with docker installed.

A simple reproduction state is : 

```
pull postgres docker:
  docker_image.present:
    - name: postgres:9-alpine
    - force: True
```

### Steps to Reproduce Issue
(Include debug logs if possible and relevant.)

1) Install salt, docker
2) Create a state as shown before.
3) run `salt your_minion state.apply the_state`

This gives

```
[vagrant@localhost ~]$ salt "*" state.apply test -l debug
[DEBUG   ] Reading configuration from /etc/salt/master
[DEBUG   ] Using cached minion ID from /etc/salt/minion_id: 172.17.0.1
[DEBUG   ] Missing configuration file: /home/vagrant/.saltrc
No permissions to access "/var/log/salt/master", are you running as the correct user?
[vagrant@localhost ~]$ aauso^C
[vagrant@localhost ~]$ sudo !!
sudo salt "*" state.apply test -l debug
[DEBUG   ] Reading configuration from /etc/salt/master
[DEBUG   ] Using cached minion ID from /etc/salt/minion_id: 172.17.0.1
[DEBUG   ] Missing configuration file: /root/.saltrc
[DEBUG   ] Configuration file path: /etc/salt/master
[WARNING ] Insecure logging configuration detected! Sensitive data may be logged.
[DEBUG   ] Reading configuration from /etc/salt/master
[DEBUG   ] Using cached minion ID from /etc/salt/minion_id: 172.17.0.1
[DEBUG   ] Missing configuration file: /root/.saltrc
[DEBUG   ] MasterEvent PUB socket URI: /var/run/salt/master/master_event_pub.ipc
[DEBUG   ] MasterEvent PULL socket URI: /var/run/salt/master/master_event_pull.ipc
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for (u'/etc/salt/pki/master', '172.17.0.1_master', u'tcp://127.0.0.1:4506', u'clear')
[DEBUG   ] Initializing new IPCClient for path: /var/run/salt/master/master_event_pub.ipc
[DEBUG   ] LazyLoaded local_cache.get_load
[DEBUG   ] Reading minion list from /var/cache/salt/master/jobs/02/7a2a9a4f342b0ca3d6d08972a4801260a8d1f810555af991496815986d65e3/.minions.p
[DEBUG   ] get_iter_returns for jid 20170815084444942374 sent to set(['172.17.0.1']) will timeout at 08:44:49.989009
[DEBUG   ] Checking whether jid 20170815084444942374 is still running
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for (u'/etc/salt/pki/master', '172.17.0.1_master', u'tcp://127.0.0.1:4506', u'clear')
[DEBUG   ] Passing on saltutil error. Key 'u'retcode' missing from client return. This may be an error in the client.
[DEBUG   ] Checking whether jid 20170815084444942374 is still running
[DEBUG   ] Initializing new AsyncZeroMQReqChannel for (u'/etc/salt/pki/master', '172.17.0.1_master', u'tcp://127.0.0.1:4506', u'clear')
[DEBUG   ] Passing on saltutil error. Key 'u'retcode' missing from client return. This may be an error in the client.
[DEBUG   ] jid 20170815084444942374 return from 172.17.0.1
[DEBUG   ] return event: {'172.17.0.1': {u'jid': '20170815084444942374', u'retcode': 2, u'ret': {'docker_image_|-pull postgres docker_|-postgres:9-alpine_|-present': {'comment': "Image 'postgres:9-alpine' could not be pulled", 'name': 'postgres:9-alpine', 'start_time': '08:44:45.158444', 'result': False, 'duration': 22513.142, '__run_num__': 0, '__sls__': 'test', 'changes': {'Status': 'Image is up to date for docker.io/postgres:9-alpine', 'Time_Elapsed': 22.496347904205322}, '__id__': 'pull postgres docker'}}, u'out': 'highstate'}}
[DEBUG   ] LazyLoaded highstate.output
[DEBUG   ] LazyLoaded nested.output

172.17.0.1:
----------
          ID: pull postgres docker
    Function: docker_image.present
        Name: postgres:9-alpine
      Result: False
     Comment: Image 'postgres:9-alpine' could not be pulled
     Started: 08:44:45.158444
    Duration: 22513.142 ms
     Changes:   
              ----------
              Status:
                  Image is up to date for docker.io/postgres:9-alpine
              Time_Elapsed:
                  22.4963479042

Summary for 172.17.0.1
------------
Succeeded: 0 (changed=1)
Failed:    1
------------
Total states run:     1
Total run time:  22.513 s

``` 

### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)
```
Salt Version:
           Salt: 2017.7.0-444-gde88937
 
Dependency Versions:
           cffi: 1.9.1
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: 2.4.2
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.9.6
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: 2.14
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.13 (default, Jun 26 2017, 10:20:05)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.1
            ZMQ: 4.1.6
 
System Versions:
           dist: fedora 26 Twenty Six
         locale: UTF-8
        machine: x86_64
        release: 4.11.8-300.fc26.x86_64
         system: Linux
        version: Fedora 26 Twenty Six
```

Thank you !

**Comment by garethgreenaway** [comment]
@terminalmage Thoughts on this one?

**Comment by terminalmage** [comment]
I can't reproduce this on my laptop running Arch. Going to test on Fedora 26 now.

**Comment by terminalmage** [comment]
OK, I can reproduce on Fedora 26. It seems that Docker on Fedora 26 includes the registry URL before the image name:

```
[root@81893668f0ec ~]# salt-call --local docker.list_tags
local:
    - docker.io/busybox:latest
    - docker.io/postgres:9-alpine
```

I haven't seen this notation before for images that don't come from a custom repo. When I switched the SLS to use the image name `docker.io/postgres:9-alpine`, it worked.

I'm working on adjusting the state to be more tolerant of variances like this between different releases of Docker, but in the meantime, using `docker.io/postgres:9-alpine` should work for you.

**Comment by terminalmage** [comment]
This may be more complicated that I thought though, because not all platforms seem to be including the registry URL for images from the Docker Hub, and this has to work for both.

**Comment by terminalmage** [comment]
OK, this should be fixed now via #43176. The differences in image naming in `docker_container.present` states should be resolved by #43155. Both fixes should be in 2017.7.2.
